src/main/java/com/cleanroommc/groovyscript/helper/recipe/AbstractRecipeBuilder.java

```diff
@@ -162,12 +165,22 @@ public void validateFluids(GroovyLog.Msg msg, int minFluidInput, int maxFluidInp
         msg.add(fluidOutput.size() < minFluidOutput || fluidOutput.size() > maxFluidOutput, () -> getRequiredString(minFluidOutput, maxFluidOutput, "fluid output") + ", but found " + fluidOutput.size());
     }
 
+    @GroovyBlacklist
+    protected int getMaxInput() {
+        return Integer.MAX_VALUE;
+    }
+
     @GroovyBlacklist
     public void validateItems(GroovyLog.Msg msg, int minInput, int maxInput, int minOutput, int maxOutput) {
         input.trim();
         output.trim();
         msg.add(input.size() < minInput || input.size() > maxInput, () -> getRequiredString(minInput, maxInput, "item input") + ", but found " + input.size());
         msg.add(output.size() < minOutput || output.size() > maxOutput, () -> getRequiredString(minOutput, maxOutput, "item output") + ", but found " + output.size());
+        if (GroovyScriptConfig.compat.checkInputStackCounts) {
+            int maxAmountProvided = input.stream().filter(Objects::nonNull).mapToInt(IResourceStack::getAmount).max().orElse(0);
+            int maxAmountAllowed = getMaxInput();
+            msg.add(maxAmountProvided > maxAmountAllowed, "Expected stack sizes of inputs = {} or less, got {}", maxAmountAllowed, maxAmountProvided);
+        }
```

Comment:
i think that iterating through each ingredient and adding a logging message for each one would be better

```suggestion
        if (GroovyScriptConfig.compat.checkInputStackCounts) {
            for (var ingredient : input) {
                msg.add(IngredientHelper.overMaxSize(ingredient, getMaxInput()), "Expected a stack size of less than or equal to {}, yet the stack size for ingredient {} was {}", getMaxInput(), ingredient.toString(), ingredient.getAmount());
            }
        }
```